src/algorithms/GenericFunctions.jl

```diff
@@ -466,6 +470,8 @@ function factor_squarefree(a)
    throw(NotImplementedError(:factor_squarefree, a))
 end
 
+_implements(::Type{T}, f::typeof(factor_squarefree)) where {T} = _implements_directly(T, f)
```

Comment:
We could remove these methods if we removed the default method that only consists of `throw(NotImplementedError(...))` anyway... Why do these methods exist anyway? Perhaps the error is considered more helpful for users?

Reply:
Yes

Reply:
OK, that's what I thought (and I kinda agree with the assessment, too). I can add a comment to these functions that explains why we have them.